pkg/secrets/config.go

```diff
@@ -24,9 +24,12 @@ type SecretManagerType string
 const (
 	SecretManagerTypeAWSSecretsManager    SecretManagerType = "AWS_SECRETS_MANAGER"
 	SecretManagerTypeAzureKeyVault        SecretManagerType = "AZURE_KEY_VAULT"
-	SecretManagerTypeGoogleSecretManager  SecretManagerType = "GOOGLE_SECRET_MANAGER"
 	SecretManagerTypeGoogleHashiCorpVault SecretManagerType = "HASHICORP_VAULT"
-	SecretManagerTypeNoop                 SecretManagerType = "NOOP"
+	SecretManagerTypeGoogleSecretManager  SecretManagerType = "GOOGLE_SECRET_MANAGER"
+	SecretManagerTypeInMemory             SecretManagerType = "IN_MEMORY"
+
+	// SecretManagerTypeNoop is deprecated. Use SecretManagerTypeInMemory.
+	SecretManagerTypeNoop SecretManagerType = "NOOP"
```

Comment:
I think it would be best to do a hard drop here since we made it impossible to create one.

then anyone using this as a lib would get a compile error instead of a runtime error

Reply:
Done.